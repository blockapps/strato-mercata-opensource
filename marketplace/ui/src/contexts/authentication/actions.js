import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  check: 'auth/check',
  checkSuccessful: 'auth/check_successful',
  checkFailed: 'auth/check_failed',
  checkFailedInternalServerError: 'auth/check_failed_internal_server_error',
  fetchUsers: 'FETCH_USERS',
  fetchUsersSuccessful: 'FETCH_USERS_SUCCESSFUL',
  fetchUsersFailed: 'FETCH_USERS_FAILED',
  logoutCheck: 'auth/logout_check',
  logoutSuccessful: 'auth/logout_success',
  logoutFailed: 'auth/logout_failed',
};

const actions = {
  check: async (dispatch) => {
    dispatch({ type: actionDescriptors.check });
    try {
      let response = await fetch(`${apiUrl}/users/me`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        {
          dispatch({
            type: actionDescriptors.checkFailedInternalServerError,
            error: body.error,
          });
          return;
        }
      } else if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.checkFailed,
          payload: body.error.loginUrl,
        });
        return;
      } else if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.checkSuccessful,
          payload: { ...body.data },
        });
        return;
      }
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    }
  },
  fetchUsers: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchUsers });
    try {
      const response = await fetch(`${apiUrl}/users`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchUsersSuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchUsersFailed,
          error: 'Unauthorized while fetching users',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    }
  },
  logout: async (dispatch) => {
    dispatch({ type: actionDescriptors.logoutCheck });
    try {
      const response = await fetch(`${apiUrl}/authentication/logout`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        window.location.href = body.data.logoutUrl;
        localStorage.removeItem('loginCount');
        dispatch({
          type: actionDescriptors.logoutSuccessful,
          payload: body.data.logoutUrl,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.logoutFailed,
          error: 'Unauthorized while logging out',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    }
  },
};

export { actionDescriptors, actions };
