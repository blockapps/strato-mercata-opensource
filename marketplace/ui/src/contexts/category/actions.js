import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  createCategory: 'create_category',
  createCategorySuccessful: 'create_category_successful',
  createCategoryFailed: 'create_category_failed',
  fetchCategory: 'fetch_categorys',
  fetchCategorySuccessful: 'fetch_category_successful',
  fetchCategoryFailed: 'fetch_category_failed',
  resetMessage: 'reset_message',
  setMessage: 'set_message',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  createCategory: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.createCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.createCategorySuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Category created successfully', true);
        return true;
      }

      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
    }
  },

  fetchCategories: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.GET,
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchCategorySuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Error while fetching category',
        });
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Unauthorized while fetching categories',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: body.error,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: 'Error while fetching category',
      });
    }
  },
};

export { actionDescriptors, actions };
