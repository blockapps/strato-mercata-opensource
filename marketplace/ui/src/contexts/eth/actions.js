import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  resetMessage: 'reset_message',
  setMessage: 'set_message',
  fetchETHSTAddress: 'fetch_ethst_address',
  fetchETHSTAddressSuccessful: 'fetch_ethst_address_successful',
  fetchETHSTAddressFailed: 'fetch_ethst_address_failed',
  addHash: 'add_hash',
  addHashSuccessful: 'add_hash_successful',
  addHashFailed: 'add_hash_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  fetchETHSTAddress: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchETHSTAddress });
    try {
      let response = await fetch(`${apiUrl}/tokens/address`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });

      const body = await response.json();
      if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressFailed,
          payload: 'Error while fetching ETHST address',
        });
        return null;
      }

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressSuccessful,
          payload: body?.data,
        });
        return body.data;
      }

      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    }
  },

  addHash: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.addHash });

    try {
      const response = await fetch(`${apiUrl}/tokens/addHash`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.addHashSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Hash added successfully', true);
        return true;
      } else if (response.status === RestStatus.CONFLICT) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: body.error.message,
        });
        actions.setMessage(dispatch, body.error.message);
        return false;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Error while adding Hash',
        });
        actions.setMessage(dispatch, 'Error while adding Hash');
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Unauthorized while adding Hash',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.addHashFailed,
        error: body.error,
      });
      actions.setMessage(dispatch, body.error);
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.addHashFailed,
        error: 'Error while adding Hash',
      });
      actions.setMessage(dispatch, 'Error while adding Hash');
    }
  },
};

export { actionDescriptors, actions };
