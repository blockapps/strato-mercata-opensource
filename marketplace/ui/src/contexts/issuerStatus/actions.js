import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  setMessage: 'set_message',
  resetMessage: 'reset_message',
  requestReview: 'request_review',
  requestReviewSuccessful: 'request_review_success',
  requestReviewFailed: 'request_review_failed',
  authorizeIssuer: 'authorize_issuer',
  authorizeIssuerSuccessful: 'authorize_issuer_success',
  authorizeIssuerFailed: 'athorize_issuer_failed',
  deauthorizeIssuer: 'deauthorize_issuer',
  deauthorizeIssuerSuccessful: 'deauthorize_issuer_success',
  deauthorizeIssuerFailed: 'deathorize_issuer_failed',
  modifyAdmin: 'modify_admin',
  modifyAdminSuccessful: 'modify_admin_successful',
  modifyAdminFailed: 'modify_admin_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },
  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },
  requestReview: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.requestReview });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/requestReview`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status == RestStatus.OK) {
        dispatch({ type: actionDescriptors.requestReviewSuccessful });
        actions.setMessage(dispatch, 'Successfully requested review', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.requestReviewFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (e) {
      dispatch({ type: actionDescriptors.requestReviewFailed });
      actions.setMessage(
        dispatch,
        'Error occurred while requesting review: ' + e.message,
        false
      );
    }
  },
  authorizeIssuer: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.authorizeIssuer });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/authorizeIssuer`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.authorizeIssuerSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Successfully authorized issuer', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.authorizeIssuerFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.authorizeIssuerFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
  deauthorizeIssuer: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.deauthorizeIssuer });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/deauthorizeIssuer`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.deauthorizeIssuerSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Successfully deauthorized issuer', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.deauthorizeIssuerFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.deauthorizeIssuerFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
  modifyAdmin: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.modifyAdmin });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/admin`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.modifyAdminSuccessful,
          payload: body.data,
        });
        actions.setMessage(
          dispatch,
          "Successfully updated user's admin status",
          true
        );
        return body;
      } else {
        dispatch({ type: actionDescriptors.modifyAdminFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.modifyAdminFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
};

export { actionDescriptors, actions };
