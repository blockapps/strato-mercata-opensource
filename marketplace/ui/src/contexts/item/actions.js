import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  fetchItem: 'fetch_items',
  fetchItemSuccessful: 'fetch_item_successful',
  fetchItemFailed: 'fetch_item_failed',
  resetMessage: 'reset_message',
  setMessage: 'set_message',
  fetchSerialNumbers: 'fetch_serial_numbers',
  fetchSerialNumbersSuccessful: 'fetch_serial_numbers_success',
  fetchSerialNumbersFailed: 'fetch_serial_numbers_failed',
  fetchItemOwnershipHistory: 'fetch_item_ownership_history',
  fetchItemOwnershipHistorySuccessful:
    'fetch_item_ownership_history_successful',
  fetchItemOwnershipHistoryFailed: 'fetch_item_ownership_history_failed',
  fetchItemTransfers: 'fetch_item_transfers',
  fetchItemTransfersSuccessful: 'fetch_item_transfers_successful',
  fetchItemTransfersFailed: 'fetch_item_transfers_failed',
  fetchItemRawMaterials: 'fetch_item_raw_materials',
  fetchItemRawMaterialsSuccessful: 'fetch_item_raw_materials_successful',
  fetchItemRawMaterialsFailed: 'fetch_item_raw_materials_failed',
  setActualRawMaterials: 'set_actual_raw_materials',
  transferOwnership: 'transfer_ownership',
  transferOwnershipSuccessful: 'transfer_ownership_successful',
  transferOwnershipFailed: 'transfer_ownership_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  setActualRawMaterials: (dispatch, payload) => {
    dispatch({
      type: actionDescriptors.setActualRawMaterials,
      payload: payload,
    });
  },

  fetchSerialNumbers: async (dispatch, id) => {
    dispatch({ type: actionDescriptors.fetchSerialNumbers });

    try {
      const response = await fetch(`${apiUrl}/item?inventoryId=${id}`, {
        method: HTTP_METHODS.GET,
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchSerialNumbersSuccessful,
          payload: body.data,
        });

        return true;
      }

      dispatch({
        type: actionDescriptors.fetchSerialNumbersFailed,
        error: 'Error while fetching serial numbers',
      });
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchSerialNumbersFailed,
        error: 'Error while fetching serial numbers',
      });
    }
  },

  fetchItemOwnershipHistory: async (dispatch, id) => {
    dispatch({ type: actionDescriptors.fetchItemOwnershipHistory });

    try {
      const response = await fetch(`${apiUrl}/item/ownership/${id}`, {
        method: HTTP_METHODS.GET,
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchItemOwnershipHistorySuccessful,
          payload: body.data,
        });

        return true;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchItemOwnershipHistoryFailed,
          error: 'Unauthorized while fetching ownership history',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.fetchItemOwnershipHistoryFailed,
        error: 'Error while fetching ownership history',
      });
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchItemOwnershipHistoryFailed,
        error: 'Error while fetching ownership history',
      });
      return false;
    }
  },

  fetchItemTransfers: async (
    dispatch,
    limit,
    offset,
    ownerOrg,
    order,
    date
  ) => {
    dispatch({ type: actionDescriptors.fetchItemTransfers });

    let range;
    const end = date + 86400; // This is the end of the same day, needed for the range filter.
    if (date) {
      range = `&range[]=transferDate,${date},${end}`;
    }

    try {
      const response = await fetch(
        `${apiUrl}/item/transfers?limit=${limit}&order=transferDate.${order}&offset=${offset}&or=(oldOwnerOrganization.eq.${ownerOrg},newOwnerOrganization.eq.${ownerOrg})${range}`,
        {
          method: HTTP_METHODS.GET,
        }
      );

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchItemTransfersSuccessful,
          payload: body.data,
        });

        return true;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchItemTransfersFailed,
          error: 'Unauthorized while fetching item transfers',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.fetchItemTransfersFailed,
        error: 'Error while fetching item transfers',
      });
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchItemTransfersFailed,
        error: 'Error while fetching item transfers',
      });
      return false;
    }
  },

  fetchItemRawMaterials: async (
    dispatch,
    itemUniqueProductCode,
    itemSerialNumber
  ) => {
    dispatch({ type: actionDescriptors.fetchItemRawMaterials });

    try {
      const response = await fetch(
        `${apiUrl}/item/rawmaterials?itemUniqueProductCode=${itemUniqueProductCode}&itemSerialNumber=${itemSerialNumber}`,
        {
          method: HTTP_METHODS.GET,
        }
      );

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchItemRawMaterialsSuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.fetchItemRawMaterialsFailed,
          error: 'Error while fetching item raw materials',
        });
        return;
      }

      dispatch({
        type: actionDescriptors.fetchItemRawMaterialsFailed,
        error: body.error,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchItemRawMaterialsFailed,
        error: 'Error while fetching item raw materials',
      });
    }
  },

  fetchItem: async (dispatch, limit, offset, queryValue) => {
    const query = queryValue ? `&inventoryId=${queryValue}` : '';

    dispatch({ type: actionDescriptors.fetchItem });

    try {
      const response = await fetch(
        `${apiUrl}/item?limit=${limit}&offset=${offset}${query}`,
        {
          method: HTTP_METHODS.GET,
        }
      );

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchItemSuccessful,
          payload: body.data,
        });
        return;
      }
      dispatch({ type: actionDescriptors.fetchItemFailed, error: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.fetchItemFailed, error: undefined });
    }
  },

  transferOwnership: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.transferOwnership });

    try {
      const response = await fetch(`${apiUrl}/item/transferOwnership`, {
        method: HTTP_METHODS.PUT,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.transferOwnershipSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Items transferred successfully', true);
        return true;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.transferOwnershipFailed,
          error: 'Error while transferring Items',
        });
        actions.setMessage(dispatch, 'Error while transferring Items');
        return false;
      }

      dispatch({
        type: actionDescriptors.transferOwnershipFailed,
        error: body.error,
      });
      actions.setMessage(dispatch, 'Error while transferring Items');
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.transferOwnershipFailed,
        error: 'Error while transferring Items',
      });
      actions.setMessage(dispatch, 'Error while transferring Items');
    }
  },
};

export { actionDescriptors, actions };
