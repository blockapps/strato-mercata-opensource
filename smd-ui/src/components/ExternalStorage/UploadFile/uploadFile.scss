.upload-file {
  width: 100% !important;
}

.upload-dialog {
  width: 600px;
}

.upload-dialog .content-margin {
  margin: 10px 20px;
}

.upload-button {
  margin: 15px 0px 0px 0px;
}

.upload-form {
  margin-bottom: 0px;
}