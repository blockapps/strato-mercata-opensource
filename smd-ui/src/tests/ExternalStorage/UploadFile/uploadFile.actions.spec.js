import { openUploadModal, closeUploadModal, uploadFileRequest, uploadFileSuccess, uploadFileFailure, resetError, changeUsername } from '../../../components/ExternalStorage/UploadFile/uploadFile.actions';
import { mockFormData, error } from './mockUpload';

describe('UploadFile: actions', () => {

  describe('Modal:', () => {

    test('open', () => {
      expect(openUploadModal()).toMatchSnapshot();
    });

    test('close', () => {
      expect(closeUploadModal()).toMatchSnapshot();
    });

  });

  describe('Upload:', () => {

    test('request', () => {
      expect(uploadFileRequest(mockFormData)).toMatchSnapshot();
    });

    test('success', () => {
      let result = {
        contractAddress: '23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c',
        uri: 'https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg',
        metadata: 'widescreen is one of the most important factor'
      }

      expect(uploadFileSuccess(result)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(uploadFileFailure(error)).toMatchSnapshot();
    });

  });

  test('Reset errors', () => {
    expect(resetError()).toMatchSnapshot();
  });

  test('change username', () => {
    expect(changeUsername('tanuj55')).toMatchSnapshot();
  });

});