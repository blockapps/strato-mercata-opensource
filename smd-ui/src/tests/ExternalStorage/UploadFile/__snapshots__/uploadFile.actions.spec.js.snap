// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile: actions Modal: close 1`] = `
Object {
  "type": "CLOSE_UPLOAD_MODAL",
}
`;

exports[`UploadFile: actions Modal: open 1`] = `
Object {
  "type": "OPEN_UPLOAD_MODAL",
}
`;

exports[`UploadFile: actions Reset errors 1`] = `
Object {
  "type": "RESET_UPLOAD_ERROR",
}
`;

exports[`UploadFile: actions Upload: failure 1`] = `
Object {
  "error": "ERROR",
  "type": "UPLOAD_FILE_FAILURE",
}
`;

exports[`UploadFile: actions Upload: request 1`] = `
Object {
  "data": Object {
    "address": "23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c",
    "content": Array [
      Object {
        "preview": "blob:http://localhost:3000/5952fbb0-1598-483f-a050-2e9d8f2f95f6",
      },
    ],
    "description": "widescreen is one of the most important factor",
    "password": "password",
    "provider": "s3",
    "username": "tanuj55@mailinator.com",
  },
  "type": "UPLOAD_FILE_REQUEST",
}
`;

exports[`UploadFile: actions Upload: success 1`] = `
Object {
  "result": Object {
    "contractAddress": "23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c",
    "metadata": "widescreen is one of the most important factor",
    "uri": "https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg",
  },
  "type": "UPLOAD_FILE_SUCCESS",
}
`;

exports[`UploadFile: actions change username 1`] = `
Object {
  "name": "tanuj55",
  "type": "USERNAME_FORM_CHANGE",
}
`;
